src/state/slices/assetsSlice/selectors.ts

```diff
@@ -3,15 +3,16 @@ import { AssetNamespace, AssetReference, CAIP2, caip2, CAIP19, caip19 } from '@s
 import { Asset, ChainTypes, NetworkTypes } from '@shapeshiftoss/types'
 import cloneDeep from 'lodash/cloneDeep'
 import sortBy from 'lodash/sortBy'
+import createCachedSelector from 're-reselect'
 import { ReduxState } from 'state/reducer'
 import { createDeepEqualOutputSelector } from 'state/selector-utils'
 import { selectMarketDataIds } from 'state/slices/marketDataSlice/selectors'
 
-export const selectAssetByCAIP19 = createSelector(
+export const selectAssetByCAIP19 = createCachedSelector(
   (state: ReduxState) => state.assets.byId,
   (_state: ReduxState, CAIP19: CAIP19) => CAIP19,
   (byId, CAIP19) => byId[CAIP19] || undefined,
-)
+)((_state: ReduxState, assetId: CAIP19): CAIP19 => assetId ?? 'undefined')
```

Comment:
string is intentional - this function is returning the cache key to use, and `undefined` is invalid

Reply:
Isn't `assetId` of type `CAIP19` / `string`, and so always defined (and sometimes `''`)?

I.e should this line be `)((_state: ReduxState, assetId: CAIP19 | undefined): CAIP19 => assetId ?? 'undefined')`?

Reply:
Good question but not quite, selectors can be called with assetId | undefined, but re-reselect requires a string as a cache key. Many components do this in valid cases where an assetId is not yet available. For debugging purposes I wanted to use the string 'undefined' rather than an empty string, but we could make this 'no assetId' or something more descriptive if you think it's cleaner?

Reply:
Ah reading it again I see it's too suggestions, yeah lemme change the type sig and see what happens.